katharsis-rs/src/main/java/io/katharsis/rs/JsonApiResponseFilter.java

```diff
@@ -84,4 +89,19 @@ private boolean isResourceResponse(Object response) {
 		return singleResource || resourceList;
 	}
 
+	/**
+	 * Checks the media type of the response and the type of the response data to determine whether
+	 * the response should be wrapped with the necessary JSON API tags.
+	 * @param responseContext the context for the current response
+	 * @return <code>true</code>, if the media type is {@link JsonApiMediaType#APPLICATION_JSON_API} and the
+	 * 			response is not already in JSON API format,<br />
+	 * 			<code>false</code>, otherwise.
+	 */
+	private boolean wrapResponse(ContainerResponseContext responseContext) {
+		boolean mediaTypeJsonApi = JsonApiMediaType.APPLICATION_JSON_API_TYPE.equals(responseContext.getMediaType());
+		Object response = responseContext.getEntity();
+		boolean alreadyJsonApi = response instanceof Document || response instanceof InputStream;
```

Comment:
Since Document is a Katharsis type, this should be in JSONAPI format.
A response of type InputStream should not be wrapped, since it is unclear whether this is already JSONAPI, or not.
So, the flag could probably be named "notToBeWrapped" or something similar.